core/objects.go

```diff
@@ -352,16 +302,16 @@ func (ch Challenge) IsSane(completed bool) bool {
 // Note: This method does not update the challenge on the left side of the '.'
 func (ch Challenge) MergeResponse(resp Challenge) Challenge {
 	// Only override fields that are supposed to be client-provided
-	if len(ch.Path) == 0 {
-		ch.Path = resp.Path
-	}
-
-	if len(ch.S) == 0 {
-		ch.S = resp.S
-	}
-
+	// If "tls" is not provided, default to "true"
 	if resp.TLS != nil {
 		ch.TLS = resp.TLS
+	} else {
```

Comment:
Not sure what you mean here; could you elaborate?  In any case, it doesn't sound like an issue for this PR.